library.rb

```diff
@@ -85,6 +80,13 @@ def load_from_file(class_name)
       end
     end
 
+    def popular_in_order_by(field, counter = nil)
+      grouped_by = @orders.group_by { |e| e.send(field) }
+      sort_fields = grouped_by.keys.sort { |x,y| grouped_by[y].count <=> grouped_by[x].count }
+      return sort_fields.first(counter) if counter
+      sort_fields.select { |e| grouped_by[e].count == grouped_by[sort_fields.first].count }
```

Comment:
1.Why I use select here? Really, it is more logically to take first element as the best reader or the most popular book. In this case evident method is the next: 
sort_fields = grouped_by.keys.sort { |x,y| grouped_by[y].count <=> grouped_by[x].count }.first
BUT in  real library we can have 2 or 3 or n readers who has the same quantity of read books. In this case who is the best? Is the first of this men, second or reader with number n? I think all this people achieve the first position. And the select allow us to get names of thise really bookworms.
2.Why do I sort keys? group_by returns the Hash, where key is the field(book or reader) and value is array of orders, in which this field exists. When I take keys I get all names of users or books as unique array.